components/src/core/components/List/List.vue

```diff
@@ -268,7 +268,9 @@ export default defineComponent({
 
     const filterTitle = computed((): string => {
       const itemCount =
-        state.selectedItemIndexes.length || props.filteredTotalRecordsCount;
+        props.filteredTotalRecordsCount > 0 ?
```

Comment:
what if you just switch sides or OR operation ?
`const itemCount = props.filteredTotalRecordsCount || state.selectedItemIndexes.length`
```
0 || 10 => 10
undefined || 10 => 10
5 || 10 => 5
```